ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/query/TransactionLogsIndexer.java

```diff
@@ -60,6 +67,10 @@ public TransactionLogsIndexer(
     this.scheduler = scheduler;
   }
 
+  public IndexingStatus indexAll() {
+    return generateLogBloomCache(0, Long.MAX_VALUE);
```

Comment:
This can result in indexing running twice... perhaps `ensurePreviousSegmentsArePresent(blockchain.getChainHeadBlockNumber());`?

Reply:
Yes, good suggestion. Done.